contracts/helpers/TransferHelper.sol

```diff
@@ -42,123 +55,344 @@ contract TransferHelper is TransferHelperInterface, TokenTransferrer {
+            // the conduit `execute` selector.
+            if (
+                conduitMagicValue != ConduitInterface(conduit).execute.selector
+            ) {
+                // If the external call fails, revert with the conduit key
+                // and conduit address.
+                revert InvalidConduit(conduitKey, conduit);
+            }
+        } catch (bytes memory data) {
+            // Catch reverts from the external call to the conduit and
+            // "bubble up" the conduit's revert reason.
+            revert ConduitErrorRevertBytes(data, conduitKey, conduit);
+        } catch Error(string memory reason) {
+            // Catch reverts with a provided reason string and
+            // revert with the reason, conduit key and conduit address.
+            revert ConduitErrorRevertString(reason, conduitKey, conduit);
         }
+    }
 
-        // Return a magic value indicating that the transfers were performed.
-        magicValue = this.bulkTransfer.selector;
+    /**
+     * @notice An internal function to check if a recipient address implements
+     *         onERC721Received for a given tokenId.
+     *
+     * @param recipient The ERC721 recipient on which to call onERC721Received.
+     * @param tokenId   The ERC721 tokenId of the token being transferred.
+     */
+    function _checkERC721Receiver(address recipient, uint256 tokenId) internal {
+        // Check if recipient can receive ERC721 tokens.
+        try
+            IERC721Receiver(recipient).onERC721Received(
+                address(this),
+                msg.sender,
+                tokenId,
+                ""
+            )
+        returns (bytes4 selector) {
+            // Check if onERC721Received selector is valid.
+            if (selector != IERC721Receiver.onERC721Received.selector) {
+                // Revert if recipient cannot accept
+                // ERC721 tokens.
+                revert InvalidERC721Recipient(recipient);
+            }
+        } catch (bytes memory data) {
+            // "Bubble up" recipient's revert reason.
+            revert ERC721ReceiverErrorRevertBytes(
+                data,
+                recipient,
+                msg.sender,
+                tokenId
+            );
+        } catch Error(string memory reason) {
+            // "Bubble up" recipient's revert reason.
+            revert ERC721ReceiverErrorRevertString(
+                reason,
+                recipient,
+                msg.sender,
+                tokenId
+            );
+        }
+    }
+
+    /**
+     * @notice An internal function that reverts if the passed-in recipient
+     *         is the zero address.
+     *
+     * @param recipient The recipient on which to perform the check.
+     */
+    function _checkRecipientIsNotZeroAddress(address recipient) internal pure {
+        // Revert if the recipient is the zero address.
+        if (recipient == address(0x0)) {
```

Comment:
Is `address(0)` and `address(0x0)` the same? Is `address(0x0)` more gas optimized?

Reply:
I believe they are the same